### ISSUE TITLE
Loss decreases, KL_theta and KL_eta increase

### ISSUE THREAD

**Issue by mona-timmermann** [adjidieng/DETM#7]

**Comment by Emekaborisama** [comment]
I didn't experience this while using D-ETM

**Comment by mona-timmermann** [comment]
Have you run it using adam optimizer or another optimizer? And have you tried the datasets provided by Adji Dieng, too?

I have just just run D-ETM on the official UN and the ACL datasets provided by Adji Dieng again and still observe rising KL values pretty much right away. I have freshly downloaded all the files and run the following commands:

**on UN debates:**
_%run main.py --dataset un --data_path data_un/split_paragraph_0 --optim adam --emb_path embeddings/un/skipgram_emb_300d.txt --log_interval 500 --min_df 30 --num_topics 50 --lr 0.001 --epochs 50 --enc_drop 0.1 --mode 'train' --visualize_every 100 --emb_size 300 --rho_size 300 --train_embeddings 0 --batch_size 200 --eval_batch_size 200 --save_path 'Results/'_

<img width="957" alt="Bildschirmfoto 2020-10-06 um 16 33 56" src="https://user-images.githubusercontent.com/33896612/95215905-cc3d1500-07f1-11eb-9121-97b9824ddceb.png">


**on ACL abstracts:**

_%run main.py --dataset acl --data_path data_acl_largev --optim adam --emb_path embeddings/acl/skipgram_emb_300d.txt --log_interval 500 --min_df 100 --num_topics 50 --lr 0.0008 --epochs 50 --enc_drop 0.1 --mode 'train' --visualize_every 100 --emb_size 300 --rho_size 300 --train_embeddings 0 --batch_size 200 --eval_batch_size 200 --save_path 'Results/'_

<img width="961" alt="Bildschirmfoto 2020-10-06 um 16 37 26" src="https://user-images.githubusercontent.com/33896612/95216377-45d50300-07f2-11eb-90eb-eff71b6b5f18.png">

**Comment by sdspieg** [comment]
Mona - sorry to reach out like this here via github, but we've been trying to reach out to people who might be able to help us in reproducing Adji's findings, We have some ongoing projects in which we try to use various NLP tools (including) DETM to build more/better knowledge about various aspects of international relations AND we're writing  bunch of new funding proposals along those lines. So if you, or anybody else who catches this, is intrigued/interested in this - please do reach out to [me](https://www.linkedin.com/in/sdspieg/)...

**Comment by Emekaborisama** [comment]
I really think you need to debug because during my training process my Val PPL wasn't as high as yours and the KL ETA too.

However, i tried it using ADAM optimizer

**Comment by waahlstrand** [comment]
@mona-timmermann I have experienced the same issues. The consequence of an increasing KL divergence on θ is a drastic drop in topic quality, so it is indeed essential.

My experience is that the DETM, being an LSTM, is very sensitive to parameter choices. Try to lower the learning rate further. The following yields decreasing KL divergence on all parameters for me:

`python main.py --dataset un --data_path ../data_undebates_largev/split_paragraph_0/ --emb_path ../embeddings/un/skipgram_emb_300d.txt --min_df 10 --num_topics 50 --lr 0.0001 --epochs 400  --mode train`

(Assuming you use the original data folders).

**Comment by mona-timmermann** [comment]
@waahlstrand, thanks for your help!

I have tried lowering the learning rates both on my own dataset and the UN dataset.

_%run main.py --dataset un --data_path data_undebates_largev/split_paragraph_0/ --emb_path skipgram_emb_300d.txt --min_df 10 --num_topics 50 --lr 0.0001 --epochs 200 --mode train_ (so the same as yours except for the number of epochs, run on Google Colab) yields the following results for me. Is this not what you received after lowering the learning rate? 

![UN settings github](https://user-images.githubusercontent.com/33896612/95749519-fa13d500-0c9b-11eb-8bc8-dc541a3265bd.png)

**Comment by jfcann** [comment]
Hi all, rising KL_theta isn't, in of itself, a problem - KL_theta is the KL divergence _of_ the approximate posterior attained by variational inference _from_ the topic distribution prior (standard diagonal normal). The fact that KL_theta rises (and then presumably ~convergences) indicates that the variational inference part of the model has learnt something.
